core/lib/da_client/src/clients/gcs/mod.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    fmt,
+    fmt::{Debug, Formatter},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use zksync_config::ObjectStoreConfig;
+use zksync_da_layers::{
+    types::{DataAvailabilityError, DispatchResponse, InclusionData},
+    DataAvailabilityInterface,
+};
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::{pubdata_da::StorablePubdata, L1BatchNumber};
+
+pub(crate) struct GCSDAClient {
+    object_store: Arc<dyn ObjectStore>,
+}
+
+impl GCSDAClient {
+    pub async fn new(object_store_conf: ObjectStoreConfig) -> Self {
+        GCSDAClient {
+            object_store: ObjectStoreFactory::create_from_config(&object_store_conf).await,
+        }
+    }
+}
+
+#[async_trait]
+impl DataAvailabilityInterface for GCSDAClient {
+    async fn dispatch_blob(
```

Comment:
btw - you should have these examples in https://github.com/matter-labs/hyperchain-da/blob/main/src/clients/ ...

Reply:
That was the initial plan, but the GCSDAClient depends on the object-storage package from zksync-era
I tried importing that into `hyperchain-da`, and then importing the `hyperchain-da` back into the monorepo, but then when trying to e.g. pass the config to that struct's constructor, it resolves it as a locally-referenced package, while the `hyperchain-da` expects the one from the github's commit. So to avoid this weird case of circular dependency, I decided to keed them here